### PR TITLE
DevP2P: add snap protocol

### LINKED ISSUE #1876: devp2p: SNAP Protocol Support
Placeholder issue requested by @g11tech

### PR DESCRIPTION
**Pull request by jochem-brouwer**
Closes #1876 

TODO:
- [x] Add tests

### CODE REVIEW

File: packages/devp2p/src/protocol/protocol.ts [hunk 1]
```diff
@@ -4,9 +4,10 @@ import { EventEmitter } from 'events'
 import { devp2pDebug } from '../util'
 import { Peer, DISCONNECT_REASONS } from '../rlpx/peer'
 
-export enum EthProtocol {
+export enum EthProtocol { // What does this represent?
```
Comment (ryanio) [inline packages/devp2p/src/protocol/protocol.ts:7]: This represents the protocol running on top of RLPx. Could probably just be renamed to `Protocol`, but that may conflict with the base class? `RlpxProtocol` could also work, but not sure if we want to rename/breaking change.

**Review by deauna — APPROVED** [review]

**Comment by scorbajio** [comment]
Two tests for the SNAP protocol are proposed in #1893

File: packages/devp2p/src/protocol/protocol.ts (hunk 1 above)
Comment (jochem-brouwer) [reply packages/devp2p/src/protocol/protocol.ts:7]: We could rename this enum to `ProtocolType`?

**Review by g11tech — DISMISSED** [review]

**Comment by g11tech** [comment]
@jochem-brouwer @holgerd77 this is good to merge in my opinion

**Review by g11tech — APPROVED** [review]
LGTM!

**Comment by holgerd77** [comment]
Ok, then let's do it. 😃

**Pull request merged**